File: Swift/Advocates/Models/BlogPost.swift
//
//  BlogPost.swift
//  Advocates
//
//  Created by Michael James on 11/06/2019.
//  Copyright © 2019 Mike James. All rights reserved.
//

import Foundation

class BlogPost : NSObject {
    
    var Title : String?
    var Description : String?
    var IsFamilyFriendly : Bool?
    var PrimaryImage : String?
    var PublishedDate : NSDate?
    var Summary : String?
    var Url : String?
    var ClassType : String?
    var Identifier : String?
    
    required init(from dictionary: [AnyHashable : Any]) {
        //sdk turns dictionary to concrete Swift class
        self.Title = dictionary["title"] as? String
        self.Description = dictionary["description"] as? String
        self.IsFamilyFriendly = dictionary["isFamilyFriendly"] as? Bool
        self.PrimaryImage = dictionary["primaryImage"] as? String
        self.Summary = dictionary["summary"] as? String
        self.Url = dictionary["url"] as? String
        self.ClassType = dictionary["classType"] as? String
        self.Identifier = dictionary["identifier"] as? String

        

    }
    
   
}

File: dotNET/Advocates/Models/BlogPost.cs
using System;
using System.Reflection;
using Newtonsoft.Json;

namespace Advocates.Models
{
    public class BlogPost : BaseModel
    {
        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("isFamilyFriendly")]
        public bool IsFamilyFriendly { get; set; }

        [JsonProperty("primaryImage")]
        public PrimaryImage PrimaryImage { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("url")]
        public Uri Url { get; set; }

        [JsonProperty("summary")]
        public string Summary { get; set; }


        [JsonProperty("publishedDate")]
        public DateTime PublishedDate { get; set; }


        [JsonProperty("source")]
        public string Source { get; set; }


    }

    public partial class PrimaryImage
    {
        [JsonProperty("contentUrl")]
        public Uri ContentUrl { get; set; }
    }
}
